bare_metal_billing/s3_bucket.py

```diff
@@ -0,0 +1,33 @@
+import os
+import logging
+import functools
+
+import boto3
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@functools.lru_cache
+def get_lease_bucket():
+    try:
+        s3_resource = boto3.resource(
+            service_name="s3",
+            endpoint_url=os.environ.get(
+                "S3_ENDPOINT", "https://s3.us-east-005.backblazeb2.com"
+            ),
+            aws_access_key_id=os.environ["S3_KEY_ID"],
+            aws_secret_access_key=os.environ["S3_APP_KEY"],
+        )
+    except KeyError:
+        logger.error(
+            "Error: Please set the environment variables S3_KEY_ID and S3_APP_KEY"
+        )
+    return s3_resource.Bucket(os.environ.get("S3_BUCKET_NAME", "esi-leases"))
+
+
+def fetch_s3(s3_filepath: str) -> str:
+    local_name = os.path.basename(s3_filepath)
+    lease_bucket = get_lease_bucket()
+    lease_bucket.download_file(s3_filepath, local_name)
+    return local_name
```

Comment:
I think it would make sense for the caller to just call get_lease_bucket (rather, get_bucket(bucket_name="lease-or-whatever") and then run download_file or upload_file instead of having these helper functions. I don't see the value this helper function adds at all.